third_party/terraform/resources/resource_container_node_pool.go.erb

```diff
@@ -106,12 +106,34 @@ var schemaNodePool = map[string]*schema.Schema{
 	},
 
 <% unless version == 'ga' -%>
-			"node_locations": {
-				Type:     schema.TypeSet,
-				Optional: true,
-				Computed: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
+	"node_locations": {
+		Type:     schema.TypeSet,
+		Optional: true,
+		Computed: true,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+	},
+
+	"upgrade_settings": {
+		Type:     schema.TypeList,
+		Optional: true,
+		Computed: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
```

Comment:
We're generally trying to get out of the practice of allowing users to set empty blocks in their configs, at the expense of needing to specify a bit more information than strictly necessary. We can do that be making one of these fields Required or saying AtLeastOneOf the two must be set (see https://github.com/GoogleCloudPlatform/magic-modules/pull/2608/files for examples). Do either of those make sense here?

Reply:
Sure, I have changed these fields to be required.